src/shapes/polyline.class.ts

```diff
@@ -1,353 +1,354 @@
+import { IPoint, Point } from '../point.class';
+import { TClassProperties } from '../typedefs';
 import { makeBoundingBoxFromPoints } from '../util/misc/boundingBoxFromPoints';
 import { projectStrokeOnPoints } from '../util/misc/projectStroke';
 import { degreesToRadians } from '../util/misc/radiansDegreesConversion';
+import { toFixed } from '../util/misc/toFixed';
+import { FabricObject } from './fabricObject.class';
+import { fabricObjectDefaultValues } from './object.class';
 
-(function (global) {
-  var fabric = global.fabric || (global.fabric = {}),
-    extend = fabric.util.object.extend,
-    toFixed = fabric.util.toFixed;
+export function polyFromElement<
+  T extends {
+    new (points: IPoint[], options: any): any;
+    ATTRIBUTE_NAMES: string[];
+  }
+>(
+  klass: T,
+  element: SVGElement,
+  callback: (poly: InstanceType<T> | null) => any,
+  options = {}
+) {
+  if (!element) {
+    return callback(null);
+  }
+  const points = parsePointsAttribute(element.getAttribute('points')),
+    // we omit left and top to instruct the constructor to position the object using the bbox
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    { left, top, ...parsedAttributes } = parseAttributes(
+      element,
+      klass.ATTRIBUTE_NAMES
+    );
+  callback(
+    new klass(points || [], {
+      ...parsedAttributes,
+      ...options,
+      fromSVG: true,
+    })
+  );
+}
 
+export class Polyline extends FabricObject {
   /**
-   * Polyline class
-   * @class fabric.Polyline
-   * @extends fabric.Object
-   * @see {@link fabric.Polyline#initialize} for constructor definition
+   * Points array
+   * @type Array
+   * @default
    */
-  fabric.Polyline = fabric.util.createClass(
-    fabric.Object,
-    /** @lends fabric.Polyline.prototype */ {
-      /**
-       * Type of an object
-       * @type String
-       * @default
-       */
-      type: 'polyline',
+  points: IPoint[];
 
-      /**
-       * Points array
-       * @type Array
-       * @default
-       */
-      points: null,
+  /**
+   * WARNING: Feature in progress
+   * Calculate the exact bounding box taking in account strokeWidth on acute angles
+   * this will be turned to true by default on fabric 6.0
+   * maybe will be left in as an optimization since calculations may be slow
+   * @deprecated
+   * @type Boolean
+   * @default false
+   * @todo set default to true and remove flag and related logic
+   */
+  exactBoundingBox: boolean;
 
-      /**
-       * WARNING: Feature in progress
-       * Calculate the exact bounding box taking in account strokeWidth on acute angles
-       * this will be turned to true by default on fabric 6.0
-       * maybe will be left in as an optimization since calculations may be slow
-       * @deprecated
-       * @type Boolean
-       * @default false
-       * @todo set default to true and remove flag and related logic
-       */
-      exactBoundingBox: false,
+  private initialized: true | undefined;
```

Comment:
I typed it this way since we setting it to false by default is pointless because the value will be assigned after super is called and by then set has been called as well. The whole point of the flag is to prevent that call.
So it is undefined while calling super and true once super is ready.

Reply:
yeah we have to rework this. there is something similar also in text and just can't be. There must be a way to make the code straight.